### ISSUE TITLE
Panic if audio device has sample rate other than 44100.

### ISSUE THREAD

**Issue by VladasZ** [agourlay/ruxguitar#1]
I have this issue both on macOS and Windows.

```
[2024-07-15T12:49:23Z INFO  ruxguitar::ui::picker] Loading file: "as_i_lay_dying-elegy.gp5"
[2024-07-15T12:49:26Z INFO  ruxguitar::audio::midi_player] Toggle audio stream
thread 'main' panicked at src/audio/midi_player.rs:202:5:
assertion `left == right` failed
  left: 48000
 right: 44100
```

**Comment by agourlay** [comment]
Thank you for reporting the issue :+1: 

I will make the audio setup less dependent on my setup and accept more sample rate.

**Comment by rabarar** [comment]
had the same issue - commented out the assert on line 202 for the mean time so i can play without panicing...

**Comment by rarnhe** [comment]
same issue on MacOS Mojave, but works fine on windows 10. Both compiled with cargo debug mode.

**Comment by agourlay** [comment]
Thanks everyone for the reports!

I have created the following PR with a fix https://github.com/agourlay/ruxguitar/pull/5

Please give it a try, it should accept any sampling rate and number of output channels.

**Comment by rabarar** [comment]
Works fine!On Jul 22, 2024, at 2:34 PM, Arnaud Gourlay ***@***.***> wrote:﻿
Thanks everyone for the reports!
I have created the following PR with a fix #5
Please give it a try, it should accept any sampling rate and number of output channels.

**Comment by VladasZ** [comment]
@agourlay yeah no problems when using build from that commit. Thank you!.

**Comment by agourlay** [comment]
Thanks for testing, awesome :+1: 

@VladasZ Have you tried with your 6 channels setup as well?

**Comment by rarnhe** [comment]
compiled and running on macOS, works great!

**Comment by VladasZ** [comment]
@agourlay yes 6 channels work fine too.

**Comment by agourlay** [comment]
Fixed in [0.1.1](https://github.com/agourlay/ruxguitar/releases/tag/v0.1.1)
